src/internal/proc.js

```diff
@@ -600,9 +600,15 @@ export default function proc(
         }
 
         if(isErr) {
+          // TODO: pre-calculate these switches
+          // TODO: allow users to customize origin / raceOrigin key?
+          // TODO: is this behavior correct if typeof res !== Error?
+          if(annotateErr) { res.raceOrigin = key }
```

Comment:
How would you feel about a new symbol, then? Users would do `err[RACE_ORIGIN]` instead of `err.raceOrigin`